src/main/java/rx/internal/operators/BlockingOperatorLatest.java

```diff
@@ -49,7 +50,7 @@ private BlockingOperatorLatest() {
             @Override
             public Iterator<T> iterator() {
                 LatestObserverIterator<T> lio = new LatestObserverIterator<T>();
-                source.materialize().subscribe(lio);
+                lio.subscription = source.materialize().subscribe(lio);
```

Comment:
This set races with the dereference of L101 and may cause NPE. Besides, since it would be set to the value ```lio``` essentially (plus a transparent SafeSubscriber layer), you don't really need this field. Just call ```this.unsubscribe()``` on L101.